### PR TITLE
Implement VCVT (between floating-point and fixed-point) instruction

### LINKED ISSUE #1344: Unimplemented VCVT.S32.F32 instruction
Requires the VCVT.S32.F32 instruction

```
The instruction at 0x00000000003d94b0 (opcode 0xf2f10f70) is undefined!
```

**Description:**
```
vcvt.s32.f32 q8, q8, #0xf
```

## Required by: 
Obtained the above opcode error after testing with https://github.com/Ryujinx/Ryujinx/pull/1192, which got the below game past prior missing opcode crashes.
https://github.com/Ryujinx/Ryujinx-Games-List/issues/145

### PR DESCRIPTION
**Pull request by piyachetk**
Implemented instruction
- VCVT (between floating-point and fixed-point)

Fixes #1344

I originally implemented this instruction in #2242 but to make it easier for the maintainers to review, I've decided to split it into a separate pull request.

Note: This and VQMOVN/VQMOVUN are needed to play Limbo. VQMOVN/VQMOVUN is implemented in #2916 .

### CODE REVIEW

File: ARMeilleure/Decoders/OpCodeTable.cs [hunk 1]
```diff
@@ -825,6 +825,7 @@ static OpCodeTable()
             SetA32("<<<<11101x111000xxxx101xx1x0xxxx", InstName.Vcvt,     InstEmit32.Vcvt_FI,  OpCode32SimdCvtFI.Create); // Int to FP32.
             SetA32("111111101x1111xxxxxx101xx1x0xxxx", InstName.Vcvt,     InstEmit32.Vcvt_RM,  OpCode32SimdCvtFI.Create); // The many FP32 to int encodings (fp).
             SetA32("111100111x111011xxxx011xxxx0xxxx", InstName.Vcvt,     InstEmit32.Vcvt_V,   OpCode32SimdCmpZ.Create); // FP and integer, vector.
+            SetA32("1111001x1x>>>xxxxxxx111x0xx1xxxx", InstName.Vcvt,     InstEmit32.Vcvt_V2,  OpCode32SimdCvtFFixed.Create); // Between floating point and fixed point, vector
```
Comment (gdkchan) [inline ARMeilleure/Decoders/OpCodeTable.cs:828]: It would be nice to come up with a standard naming for those. On Arm64, the fixed-point vector conversion instructions are suffixed with `_V_Fixed`.

File: ARMeilleure/Instructions/InstEmitSimdCvt32.cs [hunk 2]
```diff
@@ -111,6 +111,35 @@ public static void Vcvt_V(ArmEmitterContext context)
...
+            var toFixed = op.Opc == 1;
+            int fracBits = op.Fbits;
+            var unsigned = op.U; // Sign of I32 (destination/source)
+
+            if (toFixed) // F32 to S32 or U32 (fixed)
+            {
+                EmitVectorUnaryOpF32(context, (op1) =>
+                {
+                    var a = context.Multiply(op1, ConstF(MathF.Pow(2f, fracBits)));
+                    var rounded = EmitUnaryMathCall(context, nameof(MathF.Truncate), a);
```
Comment (gdkchan) [inline ARMeilleure/Instructions/InstEmitSimdCvt32.cs:128]: Why is that necessary? Doesn't the cast to uint/int truncate?

Comment (jduncanator) [reply ARMeilleure/Instructions/InstEmitSimdCvt32.cs:128]: Also, this is potentially an expensive code path, as it has to call into managed from guest to execute MathF.Truncate. Would be good to confirm if it's needed or not.

**Comment by gdkchan** [comment]
Just a reminder, I'm just waiting the feedback being addressed to approve it. Thank you.

**Comment by AcK77** [comment]
Superseeded by #5343
